modules/components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/eventsimulator/core/internal/ServiceComponent.java

```diff
@@ -0,0 +1,373 @@
+        /*
+         * Get singleton instance of FileUploader
+         */
+        FileUploader fileUploader = FileUploader.getFileUploaderInstance();
+        try {
+            fileUploader.deleteFile(fileName);
+            jsonString = new Gson().toJson("File is deleted");
+        } catch (EventSimulationException e) {
+            throw new EventSimulationException("Failed file delete : " + e.getMessage());
+        }
+        return Response.ok().entity(jsonString).build();
+    }
+
+
+    /**
+     * This method produces service for feed simulation
+     * <p>
+     * For an execution plan It may have one or more input streams.
+     * this method provides the capability to simulate each input streams in different cases.
+     * such as simulate using CSV File, simulate using Random Data and simulate using
+     * database resource.
+     * </p>
+     *
+     * @param feedSimulationConfigDetails jsonString to be converted to FeedSimulationDto object from the request Json body.
+     * @return Response of completion of process
+     * <p>
+     * <pre>
+     *     curl -X POST -d '{"orderByTimeStamp" : "false","streamConfiguration":[{"simulationType" : "RANDOM_DATA_SIMULATION",
+     *     "streamName":"stream4","executionPlanName" : "planCustom","events": "5","delay": "1000","attributeConfiguration":
+     *     [{"type":"CUSTOMDATA","list": "WSO2,AAA,DDD,IBM"}]}]}' ​ http://localhost:9090/eventSimulation/feedSimulation
+     * </pre>
+     * <pre>
+     *     curl -X POST -d '{"orderByTimeStamp" : "false","streamConfiguration":[{"simulationType" : "RANDOM_DATA_SIMULATION",
+     *     "streamName":"stream5","executionPlanName" : "planPrimitive","events": "5","delay": "1000","attributeConfiguration":
+     *     [{"type": "PRIMITIVEBASED","min": "2","max": "200","length":"2"}]}]}' ​ http://localhost:9090/eventSimulation/feedSimulation
+     * </pre>
+     * <pre>
+     *     curl -X POST -d '{"orderByTimeStamp" : "false","streamConfiguration":[{"simulationType" : "RANDOM_DATA_SIMULATION",
+     *     "streamName":"stream5","executionPlanName" : "planRegx","events": "5","delay": "1000","attributeConfiguration":[
+     *     {"type": "REGEXBASED","pattern":"[+]?[0-9]*\\.?[0-9]+"}]}]}' ​ http://localhost:9090/eventSimulation/feedSimulation
+     * </pre>
+     * <p>
+     * http://localhost:9090/eventSimulation/feedSimulation
+     */
+    @POST
+    @Path("/feedSimulation")
+    public Response feedSimulation(String feedSimulationConfigDetails) {
+        String jsonString;
+        try {
+            //parse json string to FeedSimulationDto object
+            FeedSimulationDto feedSimulationConfig = EventSimulatorParser.feedSimulationParser(feedSimulationConfigDetails);
+            //start feed simulation
+            String uuid = UUID.randomUUID().toString();
+            executorMap.put(uuid, EventSimulatorPoolExecutor.newEventSimulatorPool(feedSimulationConfig, feedSimulationConfig.getNoOfParallelSimulationSources()));
```

Comment:
Isn't this exceeding 120 line break?

Reply:
This content is no-longer in product-sp repo.